### PR TITLE
go/batch-submitter: Add releases for go BSS

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -502,3 +503,29 @@ jobs:
           push: true
           tags: ethereumoptimism/replica-healthcheck:${{ needs.builder.outputs.replica-healthcheck }},ethereumoptimism/replica-healthcheck:latest
           build-args: BUILDER_TAG=${{ needs.builder.outputs.builder }}
+
+  batch-submitter-service:
+    name: Publish batch-submitter-service Version ${{ needs.canary-publish.outputs.canary-docker-tag }}
```
Comment (tynes) [inline .github/workflows/release.yml:508]: This file has typos in it, its referring to `canary-publish` which will not exist, it should be `needs.release.outputs.batch-submitter-service`

**Review by tynes — CHANGES_REQUESTED** [review]
A very common error is found in this PR - the same bug is introduced 1 in 3 times a service is added to the build pipeline. This indicates to me that this should be refactored eventually to deduplicate some of the config

**Review by cfromknecht — APPROVED** [review]
LGTM barring mark's comments 👍

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -502,3 +503,29 @@ jobs:
...
+          username: ${{ secrets.DOCKERHUB_ACCESS_TOKEN_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN_SECRET }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./ops/docker/Dockerfile.batch-submitter-service
+          push: true
+          tags: ethereumoptimism/batch-submitter-service:${{ needs.release.outputs.batch-submitter-service }}
```
Comment (tynes) [inline .github/workflows/release.yml:531]: We do publish a `latest` image for the other images so we could append `,ethereumoptimism/batch-submitter-service:latest` to the `tags` value and publish a latest image as well

**Review by tynes — APPROVED** [review]

**Pull request merged**